docs/proposals/enable-local-source-code-support.md

```diff
@@ -0,0 +1,301 @@
+- [Build Inputs Overhaul](https://github.com/shipwright-io/build/pull/652)
+
+Due to the state of the Remote Artifacts ( _exists in the API but is not yet fully functional_ ) feature and the uncertainty of the Build Inputs Overhaul proposal, this EP proposes to do the following changes in the current state of the Build API, meaning under `spec.source`.
+
+### Local Source via Bundles
+
+The Bundle mechanism has it´s origins from this [Bundles](https://github.com/mattmoor/mink/tree/master/pkg/bundles) package and it´s based on the premise that if a user is pushing a container image to a container registry, then workflows could use the same approach to push the local source code, and pull it at a later stage during the building image mechanism.
+
+The Bundle mechanism executes the following:
+
+- [1] Takes the source code from a local directory.
+- [2] Build a container image that can self-extracted on runtime.
+- [3] Push the container image with the source code into a container registry, making it available for further usages.
+- [4] The Build controller on signalization of local source code feature, will prepend a step with the bundle image generated in [3] on the Strategy steps, as a replacement for the current image that pulls the source code from git.
+
+_Note_: To bundle the source code, an entity ( _Shipwright CLI_ ) will require to run code that will take the local source code from the user and put it inside a container image.
+
+_Note_: On runtime, when a Shipwright BuildRun is executing, that container image will be pulled and should self-extract its bundle, making the source code available ( _placed on a shared volume_ ) for other containers ( _like BuildKit, Kaniko, et al._ ) to use it.
+
+_Note_: Any workflow using Shipwright that intends to provide support for local source code will require to re-use the Shipwright/CLI implementation or implement a standalone one.
+
+### API Changes
+
+The changes in the API will provide users the following functionalities:
+
+- [1] Specify from where ( _local directory_ ) to build a container image that will bundle the source code and that will self-extract its contents at runtime. Also known as `bundle` image.
+- [2] Specify an existing container image to use, for pulling the source code.
+
+This EP is merelely concentrated on [1], but will make [2] easily to implement at a later stage.
+
+The API changes should consider the following features:
+
+- **Ability to upload/download source code**: Users should be able to specify how to build a bundle image. Users should be able to specify which bundle image to use, if it already exists.
+
+- **Ability to provide means of authentication to a registry**: Users should be able to authenticate when pulling/pushing to their container registry of choice.
+
+- **Ability to clean-up images if desired**: Users should have means to delete bundle images after the application container image is build.
```

Comment:
As a side comment... I see that we want to watch over images (layers), and eventually clean up the layers users have been uploading their source-code. Here, I think it matches the abstraction layer that @ricardomaraschini is building with [Tagger](https://github.com/ricardomaraschini/tagger). As this feature evolves, I would like to bring this in consideration.

Reply:
That might be an option. I´m well aware of Tagger so it will be considered, however thats an implementation detail.  @otaviof pls ensure you approve, as you are an approver for the EP.

Reply:
Note: I still need to address Adams comments.

Reply:
> That might be an option. I´m well aware of Tagger so it will be considered, however thats an implementation detail.

Actually, was a side-note suggestion, something we should keep in the radar.

In case Tagger takes place during the implementation phase, or another specific approach to actively manage container-images, we then need to make sure it's clearly described in this EP.

> @otaviof pls ensure you approve, as you are an approver for the EP.

Sure! Please consider the latest comments.